variantlib/api.py

```diff
@@ -36,71 +36,38 @@
 ]
 
 
+def _unpack_variants_json(
+    variants_json: dict,
+) -> Generator[VariantDescription]:
```

Comment:
Same not convinced we need a generator here.
We shouldn't abuse that all over the place. It makes the code much more complicated to follow.
Especially that you're not unpacking 1-2 variants from the file. You're unpacking all of them.
I don't see the value here.